frame/nfts/src/features/freeze.rs

```diff
@@ -0,0 +1,62 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+use frame_support::pallet_prelude::*;
+
+/// Freeze functions allow to make particular items non-transferable.
+/// It's also possible to revert the setting back.
+/// An origin must have a `Freezer` role in order to call those methods.
+impl<T: Config<I>, I: 'static> Pallet<T, I> {
+	pub fn do_freeze_item(
```

Comment:
The calls should be the only public functions on a pallet AFAIK. Everything else `pub(crate)`.

Reply:
What if I want to call some function from another pallet? Afaik, the main usage of the uniques pallet is to provide the basic implementation to deal with NFTs and all the custom logic will be placed in other pallets.

Reply:
Any other pallet building on it should be able to go through the public api.

Reply:
Through traits

Reply:
Done